### PR TITLE
Use .warphome as the name for the home workspace

### PR DESCRIPTION
**Pull request by Morganamilo**
This stops is from being possible to acceidently get some one else's home workspace as part of your workspace stack.

### CODE REVIEW

File: pkg/workspace/workspace.go
Comment (TripleDogDare) [inline pkg/workspace/workspace.go:133]: Change the function comment from `i.e.` to `e.g.`

File: pkg/workspace/workspace.go
Comment (TripleDogDare) [inline pkg/workspace/workspace.go:150]: Change the function comment from `i.e.` to `e.g.`

File: pkg/workspace/workspace.go
Comment (TripleDogDare) [inline pkg/workspace/workspace.go:89]: `else` is redundant due to early return pattern.

**Review by warpfork — APPROVED** [review]
I think we all basically agreed on this already at the high level, so I'll give this a very quick LGTM.

I think I'm also going to go drop a tag now though, right before we merge this.  (We're not taking versioning super seriously at this stage of development, but this change still feels like a pretty noticeable bump.)  I'll tag a `v0.6.0` right before this, and I figure it might be reasonable to tag a `v0.7.0` fairly immediately after this.

File: pkg/workspace/workspace.go
Comment (TripleDogDare) [inline pkg/workspace/workspace.go:86]: I'm not a huge fan of `DotPath` as a name because it describes the path format rather than what it does. Something like `InternalsPath` would be more descriptive of function.

**Comment by TripleDogDare** [comment]
Got a few nits. Otherwise LGTM.

File: pkg/workspace/workspace.go
Comment (Morganamilo) [reply pkg/workspace/workspace.go:86]: Good point, I'll rename it to this.

**Comment by Morganamilo** [comment]
> I think we all basically agreed on this already at the high level, so I'll give this a very quick LGTM.
> 
> I think I'm also going to go drop a tag now though, right before we merge this. (We're not taking versioning super seriously at this stage of development, but this change still feels like a pretty noticeable bump.) I'll tag a `v0.6.0` right before this, and I figure it might be reasonable to tag a `v0.7.0` fairly immediately after this.

Okay, I'll leave it to you to merge then.

**Pull request merged**
